pipeline/minimmit.md

```diff
@@ -58,134 +60,138 @@ fn verify(c, c') -> bool;
 
 ## 7. Helpers
 
+_In every helper function, the first parameter `r` is the replica whose local state the helper inspects or updates._
+
 ```text
-// Find a valid parent to build on
-fn select_parent(v) -> (c', v') {
+// Replica `r` selects a valid parent to build on
+fn select_parent(r, v) -> (c', v') {
     let i = v - 1;
     while i >= 0 {
-        if notarization(c', i) ∈ proofs[i] {
-            // If there are multiple, pick any.
-            return (c', i);
+        if notarization(c', i) ∈ r.proofs[i] {
+            return (c', i); // If there are multiple, pick any
         }
-        if nullification(i) ∈ proofs[i] {
+        if nullification(i) ∈ r.proofs[i] {
             i -= 1;
             continue;
         }
-        return ⊥;
+        return (⊥, ⊥); // No proofs for view i, cannot proceed
     }
-    return genesis;
+    return (genesis, ⊥);
```

Comment:
In our code, we start at 1 to avoid this ⊥ notation for genesis (which we consider view 0). Wonder if it makes to do the same here?

Reply:
I think this requires more debate and will leave as-is for now (for future consideration).